components/server/src/workspace/garbage-collector.ts

```diff
@@ -124,4 +127,27 @@ export class WorkspaceGarbageCollector {
             span.finish();
         }
     }
+
+    // finds volume snapshots that have been superceded by newer volume snapshot and removes them
+    protected async deleteOutdatedVolumeSnapshots() {
+        const span = opentracing.globalTracer().startSpan("deleteOutdatedVolumeSnapshots");
+        try {
+            const workspaces = await this.workspaceDB.trace({ span }).findVolumeSnapshotWorkspacesForGC();
+            const volumeSnapshots = await Promise.all(
```

Comment:
Note: This is potentially unbound, and might blow up the whole process as we grow.
The other deletion mechanisms feature a "limit" which I think we could re-use here.

Reply:
Good point. Added limit. Reusing existing config param for this, assuming it is fine.